README.md

```diff
@@ -1,100 +1,63 @@
-This works by setting up the communication between the dApp and a child pop-up browser tab, which has a separate wallet signing context.
-The second wallet's information is then passed back to the parent dApp through an API based on _window.postMessage_.
+This package provides the tools and UI for a dApp to orchestrate a user linking one wallet to another. The linking mechanism uses the Decentralised IDentifier Solana program, and the associated [identity.com DID toolkit libraries](https://www.npmjs.com/package/@identity.com/sol-did-client).
 
-Civic Link is integrated in [civic.me](https://civic.me/). You can try it by connecting your primary wallet and linking an additional wallet to your _civic.me_ profile (in the Settings -> Wallet menu):
+![](link-wallet.png)
+
+The orchestration works by setting up opening a new 'child' browser window where the user can connect the wallet they want to link in a separate wallet signing context, without the 'parent' dApp needing to log the user out (the [solana wallet adapter](https://solana-labs.github.io/wallet-adapter/) doesn't allow connection of multiple wallets). The child browser window is initialised with GET parameters passed from the parent dApp. When the wallet linking is finished, the child window communicates this to the parent window using the _window.postMessage_ API.
+
+Note: the 'child' link-wallet component must be hosted in a separate application from the 'parent' dApp, alternatively, the civic-themed iframe can be referenced which is hosted at https://link.civic.me.
+
+The Civic Link tools can be seen in action as part of [civic.me](https://civic.me/). You can try it by connecting your primary wallet and linking an additional wallet to your _civic.me_ profile (in the Settings -> Wallet menu):
 ![](https://github.com/civicteam/civic-link/blob/main/resources/civic-link.gif)
 
 For more details: [Civic Help Center - How do I link an additional wallet?](https://support.civic.com/hc/en-us/articles/7012041417111-How-do-I-link-an-additional-wallet).
 
 ## Usage
 
-To use Civic Link, create a component using `LinkWalletWithOwnershipFlow` from `@civic/civic-link` wrapped with `WalletConnectionProvider` and `PostMessageProvider` (to listen to the link wallet events):
-```typescript
+### Link-wallet app (child)
+The link-wallet flow needs to run in a separately hosted application from the parent dApp and needs to implement an App like below:
+```
 import React from "react";
 import {
   LinkWalletInputParameters,
   WalletConnectionProvider,
   LinkWalletWithOwnershipFlow,
   PostMessageProvider,
 } from "@civic/civic-link";
-import { WalletAdapterNetwork } from "@solana/wallet-adapter-base";
 
-export function LinkWalletExampleComponent({
-  linkWalletInputParameters,
-  targetWindow,
-}: {
-  linkWalletInputParameters: LinkWalletInputParameters;
-  targetWindow: Window;
-}): React.ReactElement {
+export function MakePublicFlow({}): React.ReactElement {
```

Comment:
Should we name it as `MakePublicFlow`? I mean, making public is something from the civic.me domain, no?

Reply:
that's true, will update